src/Kdyby/Doctrine/DI/OrmExtension.php

```diff
@@ -622,8 +622,15 @@ protected function processMetadataDriver(Nette\DI\ServiceDefinition $metadataDri
 		}
 
 		if ($impl === self::ANNOTATION_DRIVER) {
+			$annotationExtensionName = self::ANNOTATION_DRIVER;
+			foreach ($this->compiler->getExtensions(AnnotationsExtension::class) as $annotationExtension) {
+				$annotationExtensionName = $annotationExtension->name;
+
+				break;
+			}
```

Comment:
Thinking about it further this might not be necessary. We can probably just autowire the annotation reader by removing this entire block and changing the lines below to this:

```php
$driver->arguments = [
    'paths' => Nette\Utils\Arrays::flatten($driver->arguments),
];
```

Can you try if that works?

Reply:
Use auto autowire is not possible. Annotation driver has not typed constructor...

```php
public function __construct($reader, $paths = null)
```

But choosing service through `'@class'` works.